requires.py

```diff
@@ -108,7 +128,10 @@ def send_instance_info(self):
     def check_ready(self):
         # My middle name is ready. No, that doesn't sound right.
         # I eat ready for breakfast.
+        was_ready = is_flag_set(self.expand_name('ready'))
         toggle_flag(self.expand_name('ready'), self.is_ready)
+        if self.is_ready and was_ready and self.is_changed:
```

Comment:
`ready.changed` seems it should be set when

```not self.is_ready() and was_ready``` or ```self.is_ready() and not was_ready```

is that the intent of `ready.changed`?

i'm confused what this flag does?

Reply:
This code was taken straight from interface-openstack-integration [here](https://github.com/juju-solutions/interface-openstack-integration/blob/aa365314041ccbd0018e7c73e0de39eed9be045f/requires.py#L78-L82)

The `ready.changed` flag indicates that the relation data has changed, while the charm was and still is ready.

Seems weird to me too but I'm trying to keep it consistent with the established pattern. *shrugs*

Reply:
i'll take it!